### DISCUSSION TITLE
ZLUDA for AMD GPU CUDA

### DISCUSSION THREAD

**Discussion by ceewanna** [roflcoopter/viseron#711] [Ideas]
I just heard there is an implementation of CUDA for AMD GPU called ZLUDA https://github.com/vosen/ZLUDA.
It sounds interesting if this could be integrated into viseron. 

Recently I bought an NVIDIA gpu just to test on viseron and was able to get it run on viserson. I was kind of happy to see this in log.
``
****** Checking for hardware acceleration platforms ******
OpenCL is available!
VA-API cannot be used
CUDA is available!
``
I am still trying to figure out why va-api can't be used. By the way I am running on debian linux.

**Comment by roflcoopter** [comment]
That certainly looks interesting!
I think it would be possible to implement it but would be hard for me to do it myself since i dont have an AMD GPU

**Reply by ceewanna** [reply]
I have no idea how much involved it is. I wish I could help. I have an RX6600 AMD GPU available during weekends.
If the steps to build up a test docker image are outlined in shell scripts, I could probably give it a try on my system.

By the way, I followed this https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html to setup an nvidia for viseron. I have no idea of every detail but the only thing I found not working for me is the piece involved crio.

**Reply by roflcoopter** [reply]
There is some information here on how to build the container(s): https://viseron.netlify.app/docs/developers/docker

You should probably make changes in `docker/amd64-cuda/Dockerfile.base` one
Then you need to rebuild the base container and the main container:

```bash
docker-compose --file ./azure-pipelines/docker-compose-build.yaml --env ./azure-pipelines/.env build amd64-cuda-base && \
docker-compose --file ./azure-pipelines/docker-compose-build.yaml --env ./azure-pipelines/.env build amd64-cuda-viseron

**Reply by ceewanna** [reply]
Trying to setup ZLUDA, almost at the end, it seems there is no driver support for debian's amdgpu rocm. Too bad.

**Reply by roflcoopter** [reply]
Too bad indeed! I guess we will have to wait a bit for that then try again
